scripts/toml_to_moha.py

```diff
@@ -0,0 +1,170 @@
+import tomllib
+import numpy as np
+from os.path import exists
+from os import makedirs
+import sys
+sys.path.insert(0, '../')
+import moha
+
+def set_defaults(input_data):
+    '''
+    Function for setting defaults in the parameter dictionary
+    loaded from toml input file.
+
+    Parameters
+        ----------
+        input_data: dict
+            dict loaded from toml input file.
+
+        Returns
+        -------
+        None
+    '''
+
+    # require that defaults.toml exists
+    required_default_paramfile = "defaults.toml"
+    if not exists(required_default_paramfile):
+        raise Exception("Default input file 'defaults.toml' is required.")
+    
+    # load defaults.toml data into default_data
+    with open(required_default_paramfile, "rb") as f:
+        default_data = tomllib.load(f)
+    f.close()
+
+    # set defaults in input_data
+    for param_type in default_data.keys():
+        # set required param type keys if not specified in input_data
+        if not param_type in input_data.keys():
+            input_data[param_type] = {}
+
+        # set required default param values if not specified in input_data
+        for param in default_data[param_type].keys():
+            if not param in input_data[param_type]:
+                input_data[param_type][param] = default_data[param_type][param]
+            # make all strings lowercase for case-insensitive comparisons
+            data_value = input_data[param_type][param]
+            if type(data_value) == str:
+                input_data[param_type][param] = data_value.lower()
+
+
+
+def build_moha_moltype_1d(data):
+    '''
+    Function that builds and returns hamiltonian object 
+    specific to the "1d" moltype.
+
+    Supported hamiltonians are: PPP, Huckel, Hubbard, and Heisenberg.
+
+    Parameters
+        ----------
+        data: dict
+            dict containing toml input data.
+
+        Returns
+        -------
+        moha.Ham
+    '''
+    # define parameters for 1d model
+    norb   = data["system"]["norb"]
+    alpha  = data["model"]["alpha"]
+    beta   = data["model"]["beta"]
+    gamma0 = data["model"]["gamma0"]
+    mu     = data["model"]["mu"]
+    J_eq   = data["model"]["J_eq"]
+    J_ax   = data["model"]["J_ax"]
+
+    # build connectivity
+    connectivity = [(f"C{i}", f"C{i + 1}", 1) for i in range(1, norb)]
+    if data["system"]["bc"] == "periodic":
+        connectivity += [(f"C{norb}", f"C{1}", 1)]
+
+    # build connectivity for spin models
+    spin_connectivity = np.array(np.eye(norb, k=1) + np.eye(norb, k=-1))
+    if data["system"]["bc"] == "periodic":
+        spin_connectivity[(0,-1)] = spin_connectivity[(-1,0)] = 1
+
+    # create and return hamiltonian object ham
+    #-- Fermion models --#
+    # PPP
+    if data["model"]["hamiltonian"] == "ppp":
+        gamma = gamma0 * np.eye(norb)
```

Comment:
we need to clearly document that input of PPP hamiltonian supports only input of the diagonal term $\gamma_{pq} = \gamma_0\cdot  \delta_{pq}$

Reply:
I have added the line:
"Only diagonal form of gamma matrix for PPP model is supported: 
gamma_pq = gamma0 * delta_pq."

to the description of build_moha_moltype_1d to address this.

Reply:
I've just realized, that 2 body term in our definition is 

$$\frac{1}{2} \sum_{p \neq q} \gamma_{p q}\left(\hat{n}_{p \alpha}+\hat{n}_{p \beta}-Q_p\right)\left(\hat{n}_{q \alpha}+\hat{n}_{q \beta}-Q_q\right)$$

so that p!= q. It seems like $\gamma_0$ is inside $U_p$
I don't think we even need this parameter to be specified.